pypa/pipx
# Added --global option to perform actions for all users (#754)

* [x] I have added an entry to `docs/changelog.md`

## Summary of changes
 Addressing issue #754. Added `--global` option to perform actions for all users. The global option is for pipx and affects all commands except `run` and `completions` by changing the base paths that are used to generate the install directories.
There are two important sets of changes. 
* One is a bare minimum implementation (first two commits).
* The other is a cleaner implementation (third commit).
  * This breaks out the paths into there own submodule since they definitely aren't constants anymore.

These changes also include new pytest tests and necessarily required changes to existing tests.

I'm perfectly happy to go with either of those or a complete rewrite if that's required to make this feature fit into this project. I'm not married to any of the implementation details.

## Test plan
Passes your CI process on github (via your unaltered actions in my fork https://github.com/haxwithaxe/pipx/actions/runs/6518200232).

### Manually tested by running
To verify the changes do what they are supposed to
```
python3 src/pipx --global environment
```

To test all the relevant commands for regressions.
```
python3 src/pipx --global ensurepath
sudo python3 src/pipx --global install pycowsay
sudo python3 src/pipx --global reinstall pycowsay
sudo python3 src/pipx --global inject pycowsay black
sudo python3 src/pipx --global uninject pycowsay black
sudo python3 src/pipx --global runpip pycowsay freeze
sudo python3 src/pipx --global list
sudo python3 src/pipx --global uninstall pycowsay
python3 src/pipx environment
python3 src/pipx ensurepath
python3 src/pipx install pycowsay
python3 src/pipx reinstall pycowsay
python3 src/pipx inject pycowsay black
python3 src/pipx uninject pycowsay black
python3 src/pipx runpip pycowsay freeze
python3 src/pipx list
python3 src/pipx uninstall pycowsay
```
